File: Design-Patterns/Visitor/Cpp/equipment_visitor.cpp
#include "equipment_visitor.h"

equipment_visitor::~equipment_visitor() = default;

void equipment_visitor::visit_equipment(std::shared_ptr<equipment> equipment)
{

}

void equipment_visitor::visit_weapon(std::shared_ptr<weapon> weapon)
{
	
}


void equipment_visitor::visit_armor(std::shared_ptr<armor> armor)
{
	
}

File: Design-Patterns/Visitor/CSharp/Visitors/EquipmentVisitor.cs
using Visitor.Equipments;

namespace Visitor.Visitors
{
    public abstract class EquipmentVisitor
    {
        public virtual void VisitEquipment(Equipment equipment)
        {

        }

        public virtual void VisitWeapon(Weapon weapon)
        {

        }

        public virtual void VisitArmor(Armor armor)
        {

        }
    }
}
